pipeline/main.py

```diff
@@ -1,279 +0,0 @@
-
-        # self.q = nn.Parameter(torch.tensor(np.random.rand(self.n_segs), requires_grad=True))
-        # self.q = nn.Softmax(dim=0)(self.q)
-
-        self.layers.append(MixHopConv(self.in_dim,
-                                      self.hid_dim,
-                                      p=self.p,
-                                      dropout=self.input_dropout,
-                                      activation=self.activation,
-                                      batchnorm=self.batchnorm))
-        
-        # Hidden layers with n - 1 MixHopConv layers
-        for i in range(self.num_layers - 2):
-            self.layers.append(MixHopConv(self.hid_dim * len(args.p),
-                                          self.hid_dim,
-                                          p=self.p,
-                                          dropout=self.layer_dropout,
-                                          activation=self.activation,
-                                          batchnorm=self.batchnorm))
-        
-        self.fc_layers = nn.Linear(self.hid_dim * len(args.p), self.out_dim, bias=False)
-
-    def forward(self, graph, feats):
-        for layer in self.layers:
-            feats = layer(graph, feats)
-
-        # output = 0
-        # for k in range(self.n_segs):
-        #     segment = feats[:, k * self.out_dim : (k + 1) * self.out_dim]
-        #     output = segment * self.q[k] + output
-        
-        feats = self.fc_layers(feats)
-
-        return feats
-
-def main(args):
-    # Step 1: Prepare graph data and retrieve train/validation/test index ============================= #
-    # Load from DGL dataset
-    if args.dataset == 'Cora':
-        dataset = CoraGraphDataset(raw_dir=args.raw_dir)
-    elif args.dataset == 'Citeseer':
-        dataset = CiteseerGraphDataset(raw_dir=args.raw_dir)
-    elif args.dataset == 'Pubmed':
-        dataset = PubmedGraphDataset(raw_dir=args.raw_dir)
-    else:
-        raise ValueError('Dataset {} is invalid.'.format(args.raw_dir))
-
-    graph = dataset[0]
-    graph = dgl.add_self_loop(graph)
-
-    # check cuda
-    if args.gpu > 0 and torch.cuda.is_available():
-        device = 'cuda:{}'.format(args.gpu)
-    else:
-        device = 'cpu'
-
-    # retrieve the number of classes
-    n_classes = dataset.num_classes
-
-    # retrieve labels of ground truth
-    labels = graph.ndata.pop('label').to(device).long()
-
-    # Extract node features
-    feats = graph.ndata.pop('feat').to(device)
-    n_features = feats.shape[-1]
-
-    # retrieve masks for train/validation/test
-    train_mask = graph.ndata.pop('train_mask')
-    val_mask = graph.ndata.pop('val_mask')
-    test_mask = graph.ndata.pop('test_mask')
-
-    train_idx = torch.nonzero(train_mask, as_tuple=False).squeeze().to(device)
-    val_idx = torch.nonzero(val_mask, as_tuple=False).squeeze().to(device)
-    test_idx = torch.nonzero(test_mask, as_tuple=False).squeeze().to(device)
-
-    graph.to(device)
```

Comment:
I think moving the graph to a different device is not an in-place operation.

Reply:
Fixed.